### ISSUE TITLE
Can't get LEFT_ALT + <

### ISSUE THREAD

**Issue by shugo** [Bill-Gray/PDCurses#25]
In Win32 console mode, the getch() familiy returns 0 if LEFT_ALT + < etc. is pressed.

With the following fix, 0x3C (<) is returned and PDC_KEY_MODIFIER_ALT is set just like with RIGHT_ALT.

```
diff --git a/win32/pdckbd.c b/win32/pdckbd.c
index b206a72..d89c9c5 100644
--- a/win32/pdckbd.c
+++ b/win32/pdckbd.c
@@ -415,7 +415,8 @@ static int _process_key_event(void)
        check for this first. */
 
     if (key && ( !(state & LEFT_ALT_PRESSED) ||
-        (state & RIGHT_ALT_PRESSED) ))
+        (state & RIGHT_ALT_PRESSED) ||
+	SP->save_key_modifiers ))
     {
         /* This code should catch all keys returning a printable
            character. Characters above 0x7F should be returned as
```

**Comment by Bill-Gray** [comment]
On 2017-02-05 20:03, Shugo Maeda wrote:
> In Win32 console mode, the getch() family returns 0 if LEFT_ALT + < etc. is pressed.
>

Thank you.  I see exactly the same behavior,  using the 'Input Test' feature
in testcurs.exe.  Your fix gets me the correct return for Left-Alt + <.

I want to do a little more testing,  because I'm not really sure why the
code wasn't like that in the first place.  You've definitely fixed a bug
here;  I'm just wondering if this creates any new ones.  (Thus far,  I
don't see that it does.)

-- Bill

**Comment by shugo** [comment]
Thanks for your quick response.

> I want to do a little more testing,  because I'm not really sure why the
> code wasn't like that in the first place.  You've definitely fixed a bug
> here;  I'm just wondering if this creates any new ones.  (Thus far,  I
> don't see that it does.)

Please note that my patch introduces an incompatible change that
0x61 ('a') is returned instead of ALT_A when save_key_modifiers is
true.
Personally, I prefer the changed behavior, but it may be better to keep
the old behavior for keys defined ALT_* macros.

**Comment by Bill-Gray** [comment]
On 2017-02-05 23:21, Shugo Maeda wrote:
> Please note that my patch introduces an incompatible change that
> 0x61 ('a') is returned instead of ALT_A when save_key_modifiers is
> true.
> Personally, I prefer the changed behavior, but it may be better to keep
> the old behavior for keys defined ALT_* macros.

    You're right.  We're really stuck with the old behavior in any case;
backward compatibility is very important.  (Which may seem strange,
given all the new things in this fork of PDCurses;  but I really have
tried to make it so that old programs could compile and run seamlessly
despite all the changes.)

    I think the Win32a GUI flavor may show the way to fix this (and to
get some useful keystrokes into Win32 console).  This part of Win32a :

https://github.com/Bill-Gray/PDCurses/blob/master/win32a/pdcscrn.c#L386

    was mostly copied from console Win32 PDCurses :

https://github.com/Bill-Gray/PDCurses/blob/master/win32/pdckbd.c#L183

    It may not look much like it at first,  but they are essentially
the same arrays.  Except that I added the ability (in Win32a) to detect
Alt-< and Alt-> from either the left or right Alt key,  and some other
key hits that were not being detected.

    I _think_ this can all be back-ported to Win32,  giving the console
flavor the same ability to detect these keys,  as well as "browser back"
and similar extended keys.  I'll give it a try and see what happens...

-- Bill

**Comment by Bill-Gray** [comment]
I've just pushed an update that fixes this issue.  I found that
it was possible to do this by simply replacing the 'kptab' and
'ext_kptab' arrays with the versions used in the Win32a (GUI)
flavor.  By so doing,  I also found that almost all of the extended
keys supported in Win32a for some time are also now supported in
console Win32 PDCurses.  The list of newly-added (to the console
version) keys is at

https://github.com/Bill-Gray/PDCurses/blob/master/curses.h#L1182

    I think there are still a few issues here with Alt-Shift-(key)
and Alt-Ctrl-(key).  But this update fixes a lot of issues and puts
on the right road to fixing the remaining issues.
